cmd/govim/oswatcher.go

```diff
@@ -0,0 +1,47 @@
+// +build !darwin
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+func newOSWatcher(gomodpath string) (*osWatcher, error) {
+	mw, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new watcher: %v", err)
+	}
+
+	eventCh := make(chan fsEvent)
+	go func() {
+		for {
+			e, ok := <-mw.Events
+			if !ok {
+				break
+			}
+			switch e.Op {
+			case fsnotify.Rename, fsnotify.Remove:
+				eventCh <- fsEvent{e.Name, fsOpRemoved}
+			case fsnotify.Create, fsnotify.Chmod, fsnotify.Write:
+				eventCh <- fsEvent{e.Name, fsOpChanged}
+			}
+		}
+		close(eventCh)
+	}()
+
+	return &osWatcher{
```

Comment:
This is all nicely readable and clean, but I think I'd prefer to define an interface and return a value of that type from `NewOSWatcher` (assuming we put this in an internal package)

Reply:
I did consider it but went for this because it went for this instead since it is easier to trace back (especially since the two different versions are separated by build tag, so there is essentially only one). And the whole "accept interfaces & return concrete types".

That's at least the reasoning behind it, I'll try it out using an interface instead and see how that ends up!